Microsoft.DotNet.Interactive/StdInChannelNotSupportedException.cs

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.DotNet.Interactive
+{
+    public class StdInChannelNotSupportedException : Exception
```

Comment:
This is a jupyter layer object, should be moved there

Reply:
I expect the sub-kernels would need to catch this exception, for example the `PowerShellKernel` when executing `PSHost.UI.ReadLine()` or `PSHost.UI.ReadLineAsSecureString()`.
But the sub kernel projects doesn't refer to `Microsoft.DotNet.Interactive.Jupyter` (the jupyter layer depends on the sub kernels), so it would be impossible for individual language kernels to catch the exception if we move the exception definition to the jupyter layer.

Reply:
I thought about just using `System.NotImplementedException`, but felt having a specific exception type would make it more accurate to handle this situation.

Reply:
If this is an expected case for some clients, then it's not really an exception. It would be better notify the user that the given operation isn't supported on the current frontend. 

I'm not sure this requires any changes at the `Microsoft.DotNet.Interactive` layer at this stage.

Reply:
>  It would be better notify the user that the given operation isn't supported on the current frontend.

Agreed. IMO, this exception is only a way to notify the sub-kernel that the stdin channel is not available, and it's the kernel's responsibility to decide what to return from the `ReadLine` method and generate the proper message to notify the user.

Reply:
I removed the `StdInChannelNotSupportedException` exception. Instead, the value of `AllowStdin` is passed to the `InputRequested` instance in `ExecuteRequestHandler`, and `InputRequested.Content` will throw `NotSupportedException` when it's not allowed to use stdin.
Please take another look to see if this is better.